### DISCUSSION TITLE
Decoupling CPIC versioning from PharmCAT versioning

### DISCUSSION THREAD

**Discussion by AnotherSimon** [PharmGKB/PharmCAT#146] [Ideas]
Dear developers,

In the near future I intend to develop a PGx application and PharmCAT seems like it will be an indispensable tool to avoid re-inventing the wheel.

I've only just started going through the documentation, but from what I gather the variant-drug association data from PharmVar/CPIC is "hard-coded" into the PharmCAT tool. That's all dandy as long as the tool is still under active development and regular releases are being pushed. But are there any plans (in the near or distant future) to decouple these two resources? Once PharmCAT has reached a certain maturity I expect that the release of new variant-drug interaction information will outpace development.

Long story short, is there a way to have PharmCAT take it's known interactions data from (a) database server(s) rather than relying on the internal version of the same data?

Cheers,
Simon

**Comment by whaleyr** [comment]
Hi Simon,

Thanks for posting this question. I'm afraid I can't give a very straight-forward answer because I'm not quite sure by what you mean in a few places. Also, I think understanding a bit more about how PharmCAT is built and maintained may answer your question.

It may help to start from the [PharmCAT Modules](https://pharmcat.org/methods/PharmCAT-Modules/) page. When you say "variant-drug association" you're talking about the "Clinical Guideline Recommendations". Those recommendations are authored by organizaitons like CPIC and the DPWG, annotated by PharmGKB, then pulled into PharmCAT. PharmVar is responsible for most of the "Allele Definitions" (although, not all of them). All of those data sources manage data in databases and are responsible for maintaining and releasing data. PharmCAT has a component called the `DataManager` that is responsible for pulling data from those sources, transforming them for use in PharmCAT, and saving it into the codebase.

So, PharmCAT does take it's known data from database servers, it just doesn't do it at runtime (which is what I assume you actually meant in your question but please correct me if I'm wrong).

We want PharmCAT to be as self-contained as possible so not requiring an accessible network connection at runtime is pretty important. Plus, we want PharmCAT to be easily auditable. If you're running version X.Y.Z of PharmCAT it will always use the same data no matter where/when you run it. The data can be easily reviewed either in the PharmCAT.jar file itself or through the repo hosting service (currently GitHub). So having the definition and annotation data "hard-coded" into the codebase is a feature, not a bug. New allele definitions and recommendation data is just as important as features and bug fixes and is actually part of the development of this tool so it will not outpace it. Plus, there is additional QA that is done by curators when new definitions and recommendations are available, and that QA needs to be done before we feel confident issuing a release to people using PharmCAT.

Plus, the big advantage for us is that many of the same people who contribute to PharmVar, PharmGKB, and CPIC also work on PharmCAT so updates stay pretty well apace.

Does that answer your question?

**Comment by markwoon** [comment]
Just want to highlight @whaleyr's points:

* PharmCAT's team includes the curators that generate the "variant-drug interaction information".  Even if we don't have any code enhancements, we regularly release to keep pace with data updates.   "Active development" on PharmCAT and CPIC are synonymous so your concerns are moot at this time.

* If you are interested in pulling in the latest data outside of PharmCAT's releases, you can run the code that's responsible for marshalling the data yourself (`DataManager`) and build your own release.

* We have no plans to enable live data updates over the network.

**Comment by AnotherSimon** [comment]
Thanks for both of your answers. The logic behind having everything bundled is more clear to me now.
